### PR TITLE
Removed vendor drivers from orchestrator, make orchestration_driver_b…

### PR DESCRIPTION
**Pull request by ashutosh-mishra**
…ase as generic driver

Need to provide metedata in glance image, if not provided, driver will assume following default values
supports_device_sharing=True,
supports_hotplug=True,
max_interfaces=5

### CODE REVIEW

**Comment by ashutosh-mishra** [comment]
@njagadish @rajendramachani

File: gbpservice/nfp/orchestrator/modules/device_orchestrator.py [hunk 1]
```diff
@@ -369,7 +367,8 @@ def _decrement_device_interface_count(self, device):
         self._update_network_function_device_db(device, device['status'])
 
     def _get_orchestration_driver(self, service_vendor):
-        return self.drivers[service_vendor.lower()]
+        return orchestration_driver.OrchestrationDriver(self.config)
```
Comment (rajendramachani) [inline gbpservice/nfp/orchestrator/modules/device_orchestrator.py:368]: A new object will be created every time _get_orchestration_driver method is invoked.
Can we just keep this in self.orchestration_driver, and return that here.

File: gbpservice/nfp/orchestrator/config_drivers/heat_driver.py [hunk 2]
```diff
@@ -676,7 +676,7 @@ def _update_node_config(self, auth_token, tenant_id, service_profile,
             LOG.error(_LE("No provider cidr availabale"))
             return None, None
```
Comment (njagadish) [inline gbpservice/nfp/orchestrator/config_drivers/heat_driver.py:677]: Don't see delete of orchestration drivers in this PR. Aren't they is separate files?

File: gbpservice/nfp/orchestrator/config_drivers/heat_driver.py [hunk 3]
```diff
@@ -676,7 +676,7 @@ def _update_node_config(self, auth_token, tenant_id, service_profile,
...
         service_type = service_profile['service_type']
-        service_vendor = service_profile['service_flavor']
+        # service_vendor = service_profile['service_flavor']
```
Comment (njagadish) [inline gbpservice/nfp/orchestrator/config_drivers/heat_driver.py]: remove this.

File: gbpservice/nfp/orchestrator/drivers/orchestration_driver_base.py [hunk 4]
```diff
@@ -25,6 +25,8 @@
 )
```
Comment (njagadish) [inline gbpservice/nfp/orchestrator/drivers/orchestration_driver_base.py:25]: Should the file name be changed too?

File: gbpservice/nfp/orchestrator/drivers/orchestration_driver_base.py [hunk 5]
```diff
@@ -39,16 +41,16 @@ def wrapped(self, *args, **kwargs):
     return wrapped
 
 
-class OrchestrationDriverBase(object):
+class OrchestrationDriver(object):
     """Generic Driver class for orchestration of virtual appliances
 
     Does not support sharing of virtual appliance for different chains
     Does not support hotplugging interface to devices
```
Comment (njagadish) [inline gbpservice/nfp/orchestrator/drivers/orchestration_driver_base.py]: Correct the comment

File: gbpservice/nfp/orchestrator/drivers/orchestration_driver_base.py [hunk 6]
```diff
@@ -39,16 +41,16 @@ def wrapped(self, *args, **kwargs):
...
     Launches the VM with all the management and data ports and a new VM
     is launched for each Network Service Instance
     """
-    def __init__(self, config, supports_device_sharing=False,
-                 supports_hotplug=False, max_interfaces=5):
+    def __init__(self, config, supports_device_sharing=True,
+                 supports_hotplug=True, max_interfaces=5):
         self.service_vendor = 'general'
```
Comment (njagadish) [inline gbpservice/nfp/orchestrator/drivers/orchestration_driver_base.py:52]: What is this general service vendor?

File: gbpservice/nfp/orchestrator/drivers/orchestration_driver_base.py [hunk 7]
```diff
@@ -180,6 +182,75 @@ def _delete_interfaces(self, device_data, interfaces,
...
+        attr_value = getattr(self, attr)
+        if vendor_data.get(attr):
+            setattr(self, attr, vendor_data[attr])
+        else:
+            LOG.info(_LI("Vendor data specified in image, doesn't contains "
+                         "%(attr)s value, proceeding with default value "
+                         "%(default)s"),
+                     {'attr': attr, 'default': attr_value})
+
+    def _update_vendor_data(self, device_data):
```
Comment (njagadish) [inline gbpservice/nfp/orchestrator/drivers/orchestration_driver_base.py]: Let us discuss on this vendor data API. I need to understand this better.

File: gbpservice/nfp/orchestrator/drivers/orchestration_driver_base.py [hunk 8]
```diff
@@ -212,6 +281,10 @@ def get_network_function_device_sharing_info(self, device_data):
         ):
             raise exceptions.IncompleteData()
 
+        self._update_vendor_data(device_data['service_details']['image_name'],
```
Comment (njagadish) [inline gbpservice/nfp/orchestrator/drivers/orchestration_driver_base.py]: Do we need to call this in multiple functions? Isn't the meta data same for a particular image?

**Comment by ashutosh-mishra** [comment]
raised another pr https://github.com/oneconvergence/group-based-policy/pull/336

**Pull request closed without merging**